### PR TITLE
build(ci): add workflow for releasing main image

### LINKED ISSUE #1155: Adding a new workflow to build a container image tagged main on PR merge
It would be useful to have a Github workflow that automatically builds a container image whenever a PR is merged into the main branch. In this way, users can play with features from the latest version of the code easily via running a GHCR image without having to build it themselves.

### PR DESCRIPTION
**Pull request by kranurag7**
GitHub workflow that will build and push a new image whenever a new commit is merged on main branch.

I added a commit on my fork's main for [testing](https://github.com/kranurag7/oras/pkgs/container/oras). 

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1155

**Please check the following list**:
- [ ]  Does the affected code have corresponding tests, e.g. unit test, E2E test?
- [ ]  Does this change require a documentation update?
- [ ]  Does this introduce breaking changes that would require an announcement or bumping the major version?
- [x]  Do all new files have an appropriate license header?

### CODE REVIEW

File: .github/workflows/release-ghcr-main.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+name: release-ghcr-main
+
+on:
+  push:
+    branches: [main]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
```
Comment (TerryHowe) [inline .github/workflows/release-ghcr-main.yml:22]: Is this pretty standard as a build image?

Comment (kranurag7) [reply .github/workflows/release-ghcr-main.yml:22]: This is the environment of the runner in which we execute our job. In this case, it's an `ubuntu-latest` machine. 
If you want the builder instance to be specific, then we can use `runs-on: ubuntu-22.04`. 
I'm in favor of keeping it specific, but in other workflow we use the latest instance, so I preferred the same. 

https://github.com/oras-project/oras/blob/9ecf303eeb2620dfe3ddf15721999d1f7e8fd4bf/.github/workflows/release-ghcr.yml#L23

**Review by qweeah — APPROVED** [review]
LGTM. 

I think what's proposed in this PR can be merged into the https://github.com/oras-project/oras/blob/main/.github/workflows/release-ghcr.yml by changing the prepare job to
```
TAG=${GITHUB_REF#refs/*/}
echo "ref=ghcr.io/${{ github.repository }}:${TAG}" >> $GITHUB_OUTPUT
```
This can be done in another PR.

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**